### PR TITLE
added createText function and Text class

### PR DESCRIPTION
**Pull request by two-ticks**
this PR allows us to do something as below
```js
  let texty = createText('Ampere Circulation Law');
  texty.position(25, 175);
  texty.fill('red');
  texty.size(35);
  texty.play();
  //play(texty); //can use this also
```
[open in web-editor](https://editor.p5js.org/radium.scientist/sketches/GzOX2Vz37)
## TL;DR
- [x] createText function
- [x] methods : `.play()`, `.size()` , `.fill()` and `.position()`

### CODE REVIEW

**Comment by JithinKS97** [comment]
@two-ticks About the interface. Does play accept any arguments so that we can make it appear in a time span. It would be good if we take that parameter in seconds.

Also, are we planning to introduce any function to make the text fade away? Rather than `play`, Should we have something like `appear` and `disappear` / `fadein` and `fadeout` ?

@nickmcintyre please tell your opinion also.

Also what if I want to make two texts appear one after the other. Will these functions let us do that also?

As we merge these code, it would be good if we also have the documentation side by side.

File: src/lib/MObject/Text.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+export class Text {
+  writeTextElement: any;
```
Comment (JithinKS97) [inline src/lib/MObject/Text.ts]: @two-ticks Rather than giving any, can we define an interface and give it here. I hope you know about interfaces in typescript.

File: src/lib/Scene/play.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * @example
+ * example for playing animation of type 'appear' for TeX object:
+ * ```js
+ * play(tex_1, 'appear', 2000);
+ * ```
+ * @experimental
+ */
+export function play( //TODO: use '...args'
+  _object: any,
+  animation_type: string = 'write',
```
Comment (JithinKS97) [inline src/lib/Scene/play.ts]: Why do we have snake case here, I think it will be better if we follow camelcase throughout

File: src/lib/MObject/Text.ts [hunk 3]
```diff
@@ -1,10 +1,15 @@
+import { play } from '../Scene/play';
+
 export class Text {
   writeTextElement: any;
   textWrapper: any;
   x: number;
   y: number;
   sentence: string;
   sizePx: number; //px
+  _stroke: string;
```
Comment (JithinKS97) [inline src/lib/MObject/Text.ts]: Since giving `stroke` directly won't cause any issues, do we still need to provide an underscore here?

File: src/lib/Scene/add.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Text } from '../MObject/Text';
+
+export function add(_object: any) {
```
Comment (JithinKS97) [inline src/lib/Scene/add.ts]: Is it possible to define an interface for object also?

**Comment by nickmcintyre** [comment]
@two-ticks this is great! A couple of questions:

Why is `reel()` async? The example sketch runs with without it.

How do you think we should approach sequencing multiple effects on the same `Text` object? The following code didn't work properly.

```javascript
function setup() {
  createCanvas(400, 400);
  scene = new Scene();
  reel();
}

function draw() {
  background(220);
}

function reel() {
  let texty = createText('Ampere Circulation Law');
  texty.position(25, 175);
  texty.fill('red');
  texty.size(35);
  texty.play('spinOut');
  //play(texty); //can use this also
  let text2 = createText('Linear Transformation');
  text2.position(25, 325);
  text2.fill('white');
  text2.size(35);
  text2.play('waveIn');
  // text2.play('waveOut'); // plays in place of waveIn
}
```

**Comment by two-ticks** [comment]
`async` has no use here. It is just for sake of consistency with previous sketches ( I keep reel async to compare `async` delays with animeJS delays in sketches). Sorry for the mess. 

> How do you think we should approach sequencing multiple effects on the same Text object? 

This seems to be a big problem. I can do this with async delays but with regular delays, the last animation overrides the previous animation and only the last animation is played. I am looking into the animeJS documentation if we can have something useful to avoid this issue. 

With async, the sketch is like this,
```js 
function setup() {
  createCanvas(400, 400);
  scene = new Scene();
  reel();
}

function draw() {
  background(220);
}

async function reel() {
  let texty = createText('Ampere Circulation Law');
  texty.position(25, 175);
  texty.fill('red');
  texty.size(35);
  texty.play('spinOut', 2, 0);
  //play(texty); //can use this also
  let text2 = createText('Linear Transformation');
  text2.position(25, 325);
  text2.fill('white');
  text2.size(35);
  await scene.delay(2);
  text2.play('waveIn', 2, 0);
  await scene.delay(2);
  text2.play('waveOut',2, 0);// plays after waveIn
}
```

Ykfwlw481X

I am looking into animeJS docs to see how to do this without async-await.

**Comment by two-ticks** [comment]
🎉 Some good news! Now we can do sequencing of multiple effects on the same `Text` object. (still buggy, results are not as good as async-await delays) 
[open in web-editor](https://editor.p5js.org/radium.scientist/sketches/LVPT38ig-)
```js
  let text1 = createText("Linear Transformation");
  text1.position(30, 175);
  text1.fill("white");
  text1.size(35);

  text1.play("fadeOut", 2);
  text1.play("fadeIn", 2, 3);
  text1.play("waveIn", 2, 6);
```
QwNhG1AMNq

@JithinKS97 I initiated documentation, will be extending to cover `Text`. Now we can take the parameter in seconds. 

## TL;DR
- [x] createText function
- [x] methods : `.play()`, `.size()` , `.fill()` and `.position()`
- [x] sequencing of multiple effects on the same `Text` object
- [x] delay and duration parameter in seconds
- [ ] documentation
- [ ] unit tests

**Review by JithinKS97 — APPROVED** [review]

File: src/lib/MObject/Text.ts (hunk 1 above)
Comment (two-ticks) [reply src/lib/MObject/Text.ts]: made it p5.Element type 

```diff
- writeTextElement: any;
+ writeTextElement!: p5.Element;
```

File: src/lib/MObject/Text.ts (hunk 3 above)
Comment (two-ticks) [reply src/lib/MObject/Text.ts]:
```diff
- _stroke: string;
+ fillColor: p5.Color;
```
